routing/pathfind.go

```diff
@@ -75,7 +75,7 @@ type ChannelHop struct {
 // included within the Sphinx packet.
 type Hop struct {
 	// Channel is the active payment channel edge that this hop will travel
-	// along.
+	// along. This is the _incoming_ channel to this hop.
```

Comment:
This is the _outgoing_ channel. The HTLC will flow on this path.

Reply:
As I understood it, a hop is an intermediate or final *node* (from rfc). 

The result of findPath is a list of channels that the htlc should flow along. First element in this list contains the first channel. From source to first intermediate hop. So if the first element in the list is the first hop, than this Channel property must be the incoming channel to the first hop?

Reply:
As I understood it `Hop` is analogous to an "edge" here, so saying it is incoming/outgoing doesn't make that much sense. However, the `ChannelHop` contains a `ChannelEdgePolicy` that  houses the `NodeID` of the node this edge leads to. So something about that can be added to the comment.

Reply:
Yes, it is in the language. According to the lightning rfc, a hop is a node. The `Hop` type is really what it should be. But `ChannelHop` might be better renamed to something else. `ChannelEdge` then probably. Not too far off, it's a policy, a target and a bandwidth estimate.